### ISSUE TITLE
Does not work with in-memory db with file persistence

### ISSUE THREAD

**Issue by rodonal** [nflaig/loopback4-migration#13]
Hello,
I have been trying to set up a simple example to use this module. Unfortunately I hit the following error:
`Cannot start the application. TypeError: Cannot convert undefined or null to object
    at Function.keys (<anonymous>)
    at Memory._findAllSkippingIncludes (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/connectors/memory.js:457:22)
    at Memory.all (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/connectors/memory.js:531:22)
    at invokeConnectorMethod (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/dao.js:172:21)
    at /home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/dao.js:1983:7
    at doNotify (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
    at doNotify (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/observer.js:157:49)
    at ObserverMixin._notifyBaseObservers (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/observer.js:180:5)
    at ObserverMixin.notifyObserversOf (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
    at ObserverMixin._notifyBaseObservers (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/observer.js:178:15)
    at ObserverMixin.notifyObserversOf (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/observer.js:155:8)
    at Function.find (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/dao.js:1981:10)
    at Function.findOne (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback-datasource-juggler/lib/dao.js:2028:8)
    at MigrationRepository.findOne (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/@loopback/repository/dist/repositories/legacy-juggler-bridge.js:311:59)
    at MigrationRepository.findLatestMigration (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback4-migration/dist/repositories/migration.repository.js:38:44)
    at MigrationService.migrate (/home/rxhe/dev/work_tests/loopback4-example-todo/node_modules/loopback4-migration/dist/services/migration.service.js:35:64)`
 
 
 Basically what I do is that I run the  official example of Loopback (https://loopback.io/doc/en/lb4/todo-tutorial.html) and try to add the migration module. I did the following steps.

- Installed the library/module as given.
- Created a "migrations" folder under /src.
- Created the migration script  6.0.9.migrations.ts.
- Implemented the up method based on the example.
- Added "this.component(MigrationComponent)" in the application.ts file.

    
    When I run npm start I get the above mentioned issue. I have created a Github repo to replicate the issue (https://github.com/rodonal/LoopBack-Migration-PlayGround)

**Comment by nflaig** [comment]
Hey @rodonal, thanks for reporting the issue!

Looks like this operation fails

https://github.com/nflaig/loopback4-migration/blob/03f330c8d7512408eb54cbbbb535b960581cc1de/src/repositories/migration.repository.ts#L62-L65

it seems to be a problem with the datasource, in-memory works, however in-memory with file persistence does not work (seems to be a bug there)

To make your code work just remove the file persistence (of course now won't persist data)

```diff
diff --git a/src/datasources/db.datasource.ts b/src/datasources/db.datasource.ts
index 16ca003..f60c4b7 100644
--- a/src/datasources/db.datasource.ts
+++ b/src/datasources/db.datasource.ts
@@ -10,7 +10,7 @@ const config = {
   name: 'db',
   connector: 'memory',
   localStorage: '',
-  file: './data/db.json',
+  // file: './data/db.json',
 };
 
 // Observe application's life cycle to disconnect the datasource when

```

I am not sure what your use case, this module is really only tested against "real" databases like posgres, mysql, mongodb, etc.

Feel free to open an issue on the [loopback-next repository](https://github.com/loopbackio/loopback-next) as this seems to be a bug in the in-memory db with file peristence as it doesn't support queries like `await this.findOne({order: ["changeNumber DESC", "appliedAt DESC"]})`.

**Comment by nflaig** [comment]
@rodonal just saw you are working at Camelot, there is an internal version of this module [`@camelot/migration`](https://gitlab1.camelot-idpro.de/camelot-cloud-platform/camelot-mod-migration) which you should probably use instead. Checkout https://gitlab1.camelot-idpro.de/camelot-cloud-platform/camelot-auth, this service makes quite extensive use of it.

**Comment by rodonal** [comment]
Hi @nflaig  Thanks for your response. I indeed had the same impression when I first encountered the problem that the datasource was causing the issue. It was just a first hands-on experiment with the library, so I don't think it would have any practical implications to give too much of efforts into this specific case.  Anyways, it would be beneficial to add some lines to the README file that the in-memory db case wont work with the module. I think we can close the issue.

P.S: Yeah I work at Camelot and thats why I was inclined to use this library. Great work btw.

**Comment by nflaig** [comment]
@rodonal [@camelot/migration](https://gitlab1.camelot-idpro.de/camelot-cloud-platform/camelot-mod-migration) I think should work, as far as I remember the `findLatestMigration` is implemented differently there without `order` filter

**Comment by nflaig** [comment]
I had more freedom updating the internal package and add semi-breaking changes more easily as I knew what services consumed it.

**Comment by nflaig** [comment]
I renamed your issue, I hope you don't mind, didn't find a good place to add this in README but people can quickly find this issue if the face the same problem
